WIP - added example for initial subs + re run on startup (#1980)
mongodb/docs-realm: source/sdk/node/examples/flexible-sync.txt

you set ``throwOnUpdate`` to true, adding a subscription with an existing name
but a different query throws an exception.

.. _node-sync-bootstrap-initial-subscriptions:

Bootstrap the Realm with Initial Subscriptions
``````````````````````````````````````````````

.. versionadded:: 10.18.0

> You must have at least one subscription before you can read from or write to the
> realm. You can bootstrap a realm with an initial subscription set when you open

Reviewer: ```suggestion
You must have at least one subscription before you can read from or write to a
realm with Flexible Sync. You can bootstrap a realm with an initial subscription set when you open
```

it with a :js-sdk:`SyncConfiguration <Realm.App.Sync.html#~SyncConfiguration>`
and Flexible Sync enabled. Pass the ``initialSubscriptions`` parameter with the
subscription queries you want to use to bootstrap the realm:

.. literalinclude:: /examples/generated/node/flexible-sync.snippet.create-initial-subscriptions-on-fs-realm.js
   :language: javascript

If your app needs to rerun this initial subscription every time the app starts,

> you can pass an additional parameter - ``rerunOnOpen``. This is a bool that 
> denotes whether the initial subscription should re-run every time the 

Reviewer: ```suggestion
include the additional boolean configuration value `rerunOnOpen: true``, which 
denotes that the initial subscription should re-run every time the 
```

app starts. You might need to do this to re-run dynamic time ranges 
or other queries that require a re-computation of static variables for the 
subscription.

> 

Reviewer: could you add an additional smaller example of `rerunOnOpen` to clearly highlight the api?

Author: Decided against this, I feel like 2 examples of the same method feel a bit excessive (given that the other methods on the page have 1 example per method), but I added a line emphasis and put it in an example block.

In this example, we don't want users to be overwhelmed by irrelevant tasks,
so we'll load only tasks due within the previous 7 days and the next 7 days.
Tasks that were due more than a week ago are no longer relevant, and tasks
that are due further out than the next week are also not relevant. With
``rerunOnOpen`` here, the query dynamically recalculates the relevant 
objects to sync based on the desired date range every time the app starts.

.. literalinclude:: /examples/generated/node/flexible-sync.snippet.rerun-initial-subscriptions-on-start-up.js
   :emphasize-lines: 37
   :language: javascript

> 

Reviewer: i think this is worth wrapping in an `.. example::` directive

you set ``throwOnUpdate`` to true, adding a subscription with an existing name
but a different query throws an exception.

.. _node-sync-bootstrap-initial-subscriptions:

Bootstrap the Realm with Initial Subscriptions
``````````````````````````````````````````````

.. versionadded:: 10.18.0

You must have at least one subscription before you can read from or write to a
realm with Flexible Sync enabled. You can bootstrap a realm with an initial subscription set when you open
it with a :js-sdk:`SyncConfiguration <Realm.App.Sync.html#~SyncConfiguration>`
and Flexible Sync enabled. Include the ``initialSubscriptions`` field into the
``SyncConfiguration``. Within the ``initialSubscriptions`` object, add an ``update``
field set to a callback in which you can subscribe to queries to bootstrap your

> application with:

Reviewer 2: I don't think we actually *need* the `with` and it reads a little awkwardly to me.

```suggestion
application:
```

.. literalinclude:: /examples/generated/node/flexible-sync.snippet.create-initial-subscriptions-on-fs-realm.js
   :language: javascript

If your realm needs to rerun this initial subscription on every initialization,

> include the additional boolean configuration value `rerunOnOpen: true``, 
> which denotes that the initial subscription should re-run every time the 

Reviewer 2: Nit: we could break this up to make the sentences a little more readable.

```suggestion
include the additional boolean configuration value `rerunOnOpen: true``. 
This denotes that the initial subscription should re-run every time the 
```
